### PR TITLE
Add init command

### PR DESCRIPTION
**Pull request by Pilopa**
## Proposed changes

Create and document a new initialization command to be used by plugin developers to set up a plugin for development of bdd e2e tests.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by eheimbuch** [comment]
Doesn't work as expected. I checkout an SCM-Manager Plugin, added the dev dependency and ran the init command as the Readme told me. Now i get following error when i run `yarn run e2e`.

```
▶ yarn run e2e
yarn run v1.22.5
$ integration-test-runner run -u scmadmin -p scmadmin -O ../../../build/target/ -d ./src/test/e2e
internal/modules/cjs/loader.js:834
  throw err;
  ^

Error: Cannot find module 'lru-cache'
Require stack:
- /home/edi/Projects/integration-test-runner/node_modules/semver/classes/range.js
- /home/edi/Projects/integration-test-runner/node_modules/semver/classes/comparator.js
- /home/edi/Projects/integration-test-runner/node_modules/semver/index.js
- /home/edi/Projects/integration-test-runner/src/collect-releases.js
- /home/edi/Projects/integration-test-runner/cmds/collect.js
- /home/edi/Projects/integration-test-runner/node_modules/yargs/build/index.cjs
- /home/edi/Projects/integration-test-runner/node_modules/yargs/yargs
- /home/edi/Projects/integration-test-runner/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/edi/Projects/integration-test-runner/node_modules/semver/classes/range.js:187:13)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:903:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/edi/Projects/integration-test-runner/node_modules/semver/classes/range.js',
    '/home/edi/Projects/integration-test-runner/node_modules/semver/classes/comparator.js',
    '/home/edi/Projects/integration-test-runner/node_modules/semver/index.js',
    '/home/edi/Projects/integration-test-runner/src/collect-releases.js',
    '/home/edi/Projects/integration-test-runner/cmds/collect.js',
    '/home/edi/Projects/integration-test-runner/node_modules/yargs/build/index.cjs',
    '/home/edi/Projects/integration-test-runner/node_modules/yargs/yargs',
    '/home/edi/Projects/integration-test-runner/cli.js'
  ]
}
```

File: cmds/init.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        throw new Error(
+            "Package name does not match plugin pattern ('@scm-manager/scm-mypluginname-plugin')"
+        );
+    }
+    const pluginName = pluginNameMatch[1];
+    if (!packageContent.scripts) {
+        packageContent.scripts = {};
+    }
+    packageContent.scripts.e2e =
+        "integration-test-runner run -u scmadmin -p scmadmin -O ../../../build/target/ -d ./src/test/e2e";
```
Comment (eheimbuch) [inline cmds/init.js:35]: Is this path really correct? Why `../../../build/target/`? Doesn't this write the results outside of my plugin repository?

**Review by eheimbuch — CHANGES_REQUESTED** [review]

File: cmds/init.js (hunk 1 above)
Comment (Pilopa) [reply cmds/init.js:35]: That is actually "caused" by how cypress handles this configuration. The path is relative to the e2e directory.

**Comment by Pilopa** [comment]
Unfortunately, putting the dependency into the `package.json` AND using `yarn link` seems to cause problems. Thats why I tested with `yarn link` but without putting it into the `package.json`. We should retest the workflow outlined in the readme once we published a version to `npmjs` that we can actually install.

**Pull request merged**